language/stdlib/modules/LibraAccount.move

```diff
@@ -569,73 +611,118 @@ module LibraAccount {
     /// `auth_key_prefix` | `new_account_address`, for non synthetic CoinType.
     /// Creates Preburn resource under account 'new_account_address'
     public fun create_designated_dealer<CoinType>(
-        association: &signer,
+        treasury_compliance_role: &RoleId,
+        dd_creation_capability: &Capability<CreateDesignatedDealer>,
+        parent_vasp_creation_capability: &Capability<CreateParentVASP>,
+        preburn_creation_capability: &Capability<CreatePreburn>,
```

Comment:
Similar question here: should this be combined with `CreateDesignatedDealer`?

Reply:
Happy to combine, but this was again in the "make the privileges as fine-grained as possible" -- when I look at `publish_preburn_resource` that taking a `CreateDesignatedDealer` capability seems odd, so that's why I split them.

Reply:
I agree with the principle of making the privileges as fine-grained as possible. The one thing I'm wondering is how we should handle privilege creation (i.e., publishing a resource under your account via `publish_preburn_resource` etc). That seems like the one case where it makes sense to take a role as input instead of a privilege.

For example, imagine defining (after launch) a new module that exposes privilege `X`. Privilege `X` should only be available to DD's. Should we:
(a) Do a direct write set to `Roles` to add a `CreateX` privilege + take `Capability<CreateX>` as input in `create_x`?
(b) Write `create_x` to take a `DDRole` (or  similar) as input and leave `Roles` alone?

Said differently: what do you think about a design where all `Roles::Create`* disappear and are replaced with role checks? This actually works nicely in the account creation case (I think): if you are creating a new account with role *R_creator* from role *R_created*, you would:
- Present *R_creator* to something like `Roles::new_role`. It will hand you back a `RoleId` `R_created`
- Pass *R_created* to `LibraAccount::create<relevant_account_type>`.  It can then use *R_created* to publish all the relevant resources under the freshly created account.